src/marketplace/MakeATankSaleView.js

```diff
@@ -0,0 +1,118 @@
+//@flow strict
+import * as React from 'react';
+import OptionClass from '../armory/OptionClass.js';
+import {getAllUsersTanks} from '../globalComponents/tankAPIIntegration.js';
+import {getUser} from '../globalComponents/userAPIIntegration.js';
+import {makeASale} from './marketPlaceAPIConnections.js';
+
+type Props = {||}; 
+type State = {|
+	userId: string,
+	salePrice: number,
+	tankBeingSoldId: string,
+	itemAmount: number,
+	tanksToSell: Array<OptionClass>,
+|};
+class MakeAComponentSaleView extends React.Component<Props, State> {
+
+	constructor() {
+		super();
+		this.state={
+			userId: '',
+			salePrice: 0,
+			tankBeingSoldId: '',
+			itemAmount: 0,
+			tanksToSell: [],
+		}
+		this.getUserID();
+	}
+
+	//this gets the user id
+	getUserID() : void {
+		const responsePromise = getUser();
+		responsePromise.then(
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+					console.log(data);
+					return data;
+				}
+				else {
+					const jsonObjectOfUser = data;
+					//set the users id
+					this.setState({userId:jsonObjectOfUser._id});
+					//gets the tanks that the user has currently to sell
+					this.getAllUsersTanksForSell();
+				}
+			})
+		).catch(
+			error => {
+				console.log('Couldnt connect to server!');
+				console.log(error);
+			}
+		);
+	};
+
+	//This gets all of the users tanks and then adds them to the dropdown
+	getAllUsersTanksForSell() : void {
+		const responsePromise = getAllUsersTanks();
```

Comment:
You should tank to Adam about this since you guys will both be touching this, but getAllUserTanks should return an array of Tank objects. Things get very messy if you are touching how the backend stores this stuff as JSON in multiple places because that makes it so that if we ever change how the backend stores this stuff, we have to change the entire front end as well, which is terrible design.

Reply:
Ok will talk to him about this

Reply:
Re-requesting that we don't write duplicate code to process getAllTanks

Reply:
Is this not a perfect example of shared functionality? All this method is doing is getting all tanks for a user as Tank objects, no? So this method shouldn't just call some method that gets the backend tanks and the armory should call the same method. Otherwise we have duplicate code here, and then people who are debugging this code later would get very confused as to why sometimes Tank objects have all of the data, and sometimes they only have random fields like 'tankName' and '_id'. 

If I am missing something and you actually think we need to do this a second time, we can talk about it, but it doesn't look like you addressed this at all

Reply:
For some reason, returning tanks from the tankAPIIntegration.js breaks down a lot of what we have been doing. I think this is a problem that will take some changes that I am not aware on how to do. I suggest leaving it for now, but if you want to discuss this with me, feel free to message me.